api/lib/application/authentication/oidc/oidc-controller.js

```diff
@@ -1,12 +1,14 @@
 import * as authenticationServiceRegistry from '../../../domain/services/authentication/authentication-service-registry.js';
-import * as serializer from '../../../infrastructure/serializers/jsonapi/oidc-identity-providers-serializer.js';
+import * as oidcProviderSerializer from '../../../infrastructure/serializers/jsonapi/oidc-identity-providers-serializer.js';
+import * as oidcSerializer from '../../../infrastructure/serializers/jsonapi/oidc-serializer.js';
 import { usecases } from '../../../domain/usecases/index.js';
 import { UnauthorizedError } from '../../http-errors.js';
-import * as oidcSerializer from '../../../infrastructure/serializers/jsonapi/oidc-serializer.js';
 
-const getIdentityProviders = async function (request, h) {
-  const identityProviders = usecases.getIdentityProviders();
-  return h.response(serializer.serialize(identityProviders)).code(200);
+const getIdentityProviders = async function (request, h, dependencies = { authenticationServiceRegistry }) {
+  const identityProviders = usecases.getIdentityProviders({
+    authenticationServiceRegistry: dependencies.authenticationServiceRegistry,
+  });
+  return h.response(oidcProviderSerializer.serialize(identityProviders)).code(200);
 };
```

Comment:
Normalement l'injection de dépendances classique (avant passage à ESM) est toujours disponible. Il faudrait ajouter au fichier `api/lib/domain/usecases/index.js` le service `authenticationServiceRegistry` en tant que dépendances à ajouter dans les usecases.

```suggestion
const getIdentityProviders = async function (request, h) {
  const identityProviders = usecases.getIdentityProviders();
  return h.response(oidcProviderSerializer.serialize(identityProviders)).code(200);
};
```

Reply:
Après discussion @igorissen et moi souhaitons aborder ce point en mob.

Reply:
Nous souhaitons également aborder le point de la pertinence de garder le usecase `get-identity-providers` car les OIDC Providers qui sont `isReady` sont une liste non dynamique qui peut tout à fait être initialisée à l'évaluation du module `authentication-service-registry`.

Reply:
Ce point a été abordé en mob programming comme prévu.